src/daft-local-execution/src/intermediate_ops/actor_pool_project.rs

```diff
@@ -145,23 +146,30 @@ impl IntermediateOperator for ActorPoolProjectOperator {
     #[instrument(skip_all, name = "ActorPoolProjectOperator::execute")]
     fn execute(
         &self,
-        _idx: usize,
-        input: &PipelineResultType,
-        state: &IntermediateOperatorState,
-    ) -> DaftResult<IntermediateOperatorResult> {
-        state.with_state_mut::<ActorPoolProjectState, _, _>(|state| {
-            state
+        input: &Arc<MicroPartition>,
```

Comment:
if we just clone the input right away we should take in the `Arc<MicroPartition>` since we can move the value rather than passing in a reference that is then cloned.